src/jit/codegenxarch.cpp

```diff
@@ -982,6 +982,13 @@ void CodeGen::genCodeForMul(GenTreeOp* treeNode)
             unsigned int scale = (unsigned int)(imm - 1);
             getEmitter()->emitIns_R_ARX(INS_lea, size, targetReg, rmOp->gtRegNum, rmOp->gtRegNum, scale, 0);
         }
+        else if (!requiresOverflowCheck && rmOp->isUsedFromReg() && isPow2(imm))
+        {
+            // Use shift for constant multiply when legal
+            unsigned int shiftAmount = genLog2((uint64_t)imm);
```

Comment:
Why `uint64_t` and not `size_t`? And a static_cast would be preferable to a C style cast.